src/main/java/me/zeroeightsix/kami/module/modules/misc/AutoObsidian.kt

```diff
@@ -183,198 +256,313 @@ object AutoObsidian : Module() {
+        if (mc.currentScreen is GuiShulkerBox) {
+            val container = mc.player.openContainer
+            val slot = container.inventory.subList(0, 27).indexOfFirst { it.item.id == ItemID.ENDER_CHEST.id }
+
+            if (slot != -1) {
+                InventoryUtils.inventoryClick(container.windowId, slot, 0, ClickType.QUICK_MOVE)
+                mc.player.closeScreen()
+            } else if (shulkerOpenTimer.tick(100, false)) { // Wait for maximum of 5 seconds
+                if (leaveEmptyShulkers.value && container.inventory.subList(0, 27).indexOfFirst { it.item.id != ItemID.AIR.id } == -1) {
+                    searchingState = SearchingState.PRE_MINING
+                    mc.player.closeScreen()
                 } else {
                     sendChatMessage("$chatName No ender chest was found in shulker, disabling.")
                     mc.soundHandler.playSound(PositionedSoundRecord.getRecord(SoundEvents.ENTITY_EXPERIENCE_ORB_PICKUP, 1.0f, 1.0f))
-                    this.disable()
+                    disable()
+                }
+            }
+        } else {
+            val side = EnumFacing.getDirectionFromEntityLiving(pos, mc.player)
+            val hitVecOffset = WorldUtils.getHitVecOffset(side)
+
+            lastHitVec = WorldUtils.getHitVec(pos, side)
+            rotateTimer.reset()
+
+            if (shulkerOpenTimer.tick(50)) {
+                defaultScope.launch {
+                    delay(10L)
+                    onMainThreadSafe {
+                        connection.sendPacket(CPacketPlayerTryUseItemOnBlock(pos, side, EnumHand.MAIN_HAND, hitVecOffset.x.toFloat(), hitVecOffset.y.toFloat(), hitVecOffset.z.toFloat()))
+                        player.swingArm(EnumHand.MAIN_HAND)
+                    }
                 }
-            }, delayTicks.value * 50L, TimeUnit.MILLISECONDS)
+            }
         }
     }
 
-    private fun mineBlock(pos: BlockPos, pre: Boolean) {
-        if (InventoryUtils.getSlotsHotbar(278) == null && InventoryUtils.getSlotsNoHotbar(278) != null) {
-            InventoryUtils.moveToHotbar(278, 130)
-            return
-        } else if (InventoryUtils.getSlots(0, 35, 278) == null) {
-            sendChatMessage("$chatName No pickaxe was found in inventory, disabling.")
+    private fun placeBlock(pos: BlockPos) {
+        val pair = WorldUtils.getNeighbour(pos, 1, 6.5f)
+            ?: run {
+                sendChatMessage("Can't find neighbour block")
+                return
+            }
+
+        lastHitVec = WorldUtils.getHitVec(pair.second, pair.first)
+        rotateTimer.reset()
+
+        mc.connection?.sendPacket(CPacketEntityAction(mc.player, CPacketEntityAction.Action.START_SNEAKING))
+
+        defaultScope.launch {
+            delay(10L)
+            onMainThreadSafe {
+                placeBlock(pair.second, pair.first)
+                if (NoBreakAnimation.isEnabled) NoBreakAnimation.resetMining()
```

Comment:
Maybe NoBreakAnimation should handle the if statement in it's resetMining method